### PR TITLE
WebVR: add flat torus example

### PR DESCRIPTION
**Pull request by paulmasson**
A flat torus has zero Gaussian curvature in four dimensions. Since it is a higher-dimensional object, it needs to be projected into three dimensions for visualization, and that produces an interesting object. The additional feature in this example is animation of the parameter that determines the number of lobes of the projected torus from zero, a regular looking torus, to ten. Rendering the torus as a wireframe allows more of its internal structure to be visible.

### CODE REVIEW

**Comment by paulmasson** [comment]
https://rawgit.com/paulmasson/three.js/webvr/examples/webvr_flattorus.html

File: examples/webvr_flattorus.html [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+				torus = new THREE.Mesh( geometry, material );
+				torus.position.set( 0, 1.5, - 1 );
+				scene.add( torus );
+
+				//
+
+				var light1 = new THREE.DirectionalLight( 0xffffff, 1 );
+				light1.position.set( 0, 3, - 1 );
+				light1.lookAt( 0, 1.5, - 1 );
```
Comment (WestLangley) [inline examples/webvr_flattorus.html]: `DirectionalLight ` does not support `lookAt()`.

File: examples/webvr_flattorus.html [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+				if ( count === 0 && paused ) paused = false;
+
+				for ( var j = 0 ; j <= vSteps ; j ++ ) {
+
+					for ( var i = 0 ; i <= uSteps ; i ++ ) {
+
+						f( i / uSteps, j / vSteps, vec );
+						torus.geometry.attributes.position.set( [ vec.x, vec.y, vec.z], 3 * ( i + j * ( vSteps + 1 ) ) );
+
```
Comment (WestLangley) [inline examples/webvr_flattorus.html]: You are instantiating 60 * 100 * 100 = 600000 arrays per second.

Try `.setXYZ( index, x, y, z )`, instead.

**Comment by Mugen87** [comment]
I don't think this needs to be a VR example. I vote for something like `webgl_geometry_parametric_flattorus`.

**Comment by WestLangley** [comment]
>A flat torus has zero Gaussian curvature in four dimensions. Since it is a higher-dimensional object, it needs to be projected into three dimensions for visualization

This is way over the head of most three.js users, so I do not think we need a `webgl_geometry_parametric_flattorus` example.

Granted the demo is cool-looking, but unless @mrdoob is enamored by the animation, this example may be best-suited for a math site.

**Comment by mrdoob** [comment]
@paulmasson 

How about adding this object to the `webvr_lorenzattractor` example and renaming it to `webvr_parametric_geometry?

I would also move the geometry code out of the example into `examples/js/parametric/LorenzAttractor.js` and `examples/js/parametric/FlatTorus.js`.

**Comment by paulmasson** [comment]
@mrdoob a couple comments on your recommendation:

1) The Lorenz attractor is not a parametric geometry, but the solution to a chaotic set of coupled differential equations.

2) Trying to fit both objects into a single example at a size that would allow the user to walk around both would necessitate the loss of all the detail that makes them fun. There's a lot more going on with these guys than the objects in the regular parametric geometries example.

3) Perhaps most importantly, animating both in a single example would probably crash existing hardware. For the Lorenz attractor I had to use a buffer geometry just to get it to work on an Oculus Go. Both at the same time would likely be problematic.

**Pull request closed without merging**